pkg/descheduler/strategies/topologyspreadconstraint.go

```diff
@@ -183,6 +190,16 @@ func RemovePodsViolatingTopologySpreadConstraint(
 	}
 }
 
+// hasIdenticalConstraints checks if we already had an identical TopologySpreadConstraint in namespaceTopologySpreadConstraints slice
+func hasIdenticalConstraints(newConstraint v1.TopologySpreadConstraint, namespaceTopologySpreadConstraints []v1.TopologySpreadConstraint) bool {
+	for constraint := range namespaceTopologySpreadConstraints {
```

Comment:
`for _, constraint := range namespaceTopologySpreadConstraints`

constraint is an index in here otherwise

Reply:
Good catch... I would never find this with the test being green haha